repair/row_level.cc

```diff
@@ -2277,36 +2277,86 @@ future<repair_update_system_table_response> repair_service::repair_update_system
+                [this, now, cache_disabled, &flush_time, &cache_time, &from, &req] () -> future<>  {
+                    rlogger.info("repair[{}]: Started to flush batchlog for repair_flush_hints_batchlog_request from node={}, target_nodes={}", req.repair_uuid, from, req.target_nodes);
+                    auto last_replay = _bm.local().get_last_replay();
+                    bool issue_flush = false;
+                    if (cache_disabled) {
+                        issue_flush = true;
+                        flush_time = now;
+                    } else {
+                        if (now < last_replay) {
+                            flush_time = now;
+                            utils::get_local_injector().set_parameter("repair_flush_hints_batchlog_handler", "skip_flush_use_now", fmt::to_string(flush_time));
+                        } else if (now - last_replay < cache_time) {
+                            // Skip the replay request since last_replay is already
+                            // updated since last _flush_hints_batchlog_time
+                            // update. It is fine to use last_replay for the hint
+                            // flush time because last_replay (batchlog replay
+                            // time) is smaller than now (hint flush time).
+                            flush_time = last_replay;
+                            utils::get_local_injector().set_parameter("repair_flush_hints_batchlog_handler", "skip_flush_use_last_replay", fmt::to_string(flush_time));
+                        } else {
+                            // Issue the replay so the last_replay will be updated
+                            // to bigger than now after the call.
+                            issue_flush = true;
+                            flush_time = now;
+                        }
+                    }
+                    if (issue_flush) {
+                        co_await _bm.local().do_batch_log_replay(db::batchlog_manager::post_replay_cleanup::no);
+                        utils::get_local_injector().set_parameter("repair_flush_hints_batchlog_handler", "issue_flush", fmt::to_string(flush_time));
+                    }
+                    rlogger.info("repair[{}]: Finished to flush batchlog for repair_flush_hints_batchlog_request from node={}, target_nodes={}", req.repair_uuid, from, req.target_nodes);
+                }
+            );
+        } catch (...) {
+            rlogger.warn("repair[{}]: Failed to process repair_flush_hints_batchlog_request from node={} target_hosts={}: {}",
+                    req.repair_uuid, from, req.target_nodes, std::current_exception());
+            throw;
+        }
+        co_await container().invoke_on_all([flush_time] (repair_service& rs) {
+            rs._flush_hints_batchlog_time = flush_time;
+        });
+        updated = true;
+    } else {
+        utils::get_local_injector().set_parameter("repair_flush_hints_batchlog_handler", "skip_flush", fmt::to_string(flush_time));
```

Comment:
This is a large change. Can't we eliminate some of the churn by just flipping the if and having an early return?

Reply:
If you look at the patch with --ignore-all-space, the diff will be much smaller.  Early return means we will print the finish log twice.

Reply:
Yes, I did that and it helped. Still, I usually prefer not having to do that.